ERCS/erc-7744.md

```diff
@@ -0,0 +1,101 @@
+
+Global bytecode identification is a requirement to build efficient at high scale global factories for future software distribution, that may want to rely on extensive re-use of same codebase as own dependencies.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity =0.8.20;
+
+interface ICodeIndex {
+    event Indexed(address indexed container, bytes32 indexed codeHash);
+    error alreadyExists(bytes32 id, address source);
+
+    function register(address container) external;
+
+    function get(bytes32 id) external view returns (address);
+}
+
+/**
+ * @title Byte Code Indexer Contract
+ * @notice You can use this contract to index contracts by their bytecode.
+ * @dev This allows to query contracts by their bytecode instead of addresses.
+ * @author Tim Pechersky (@Peersky)
+ */
+contract CodeIndex is ICodeIndex {
+    mapping(bytes32 => address) private index;
+
+    /**
+     * @notice Registers a contract in the index by its bytecode hash
+     * @param container The contract to register
+     * @dev `msg.codeHash` will be used
+     * @dev It will revert if the contract is already indexed
+     */
+    function register(address container) external {
+        address etalon = index[container.codehash];
+        if (etalon != address(0)) {
+            revert alreadyExists(container.codehash, etalon);
+        }
+        index[container.codehash] = container;
+        emit Indexed(container, container.codehash);
+    }
+
+    /**
+     * @notice Returns the contract address by its bytecode hash
+     * @dev returns zero if the contract is not indexed
+     * @param id The bytecode hash
+     * @return The contract address
+     */
+    function get(bytes32 id) external view returns (address) {
+        return index[id];
+    }
+}
+
+```
+
+### Deployment method
+
+The `CodeIndex` contract is deployed at: `0xc0D31d398c5ee86C5f8a23FA253ee8a586dA03Ce` using `CREATE2` via the deterministic deployer at `0x4e59b44847b379578588920ca78fbf26c0b4956c` with a salt of `0x220a70730c743a005cfd55180805d2c0d5b8c7695c5496100dcffa91c02befce`.
+
+## Rationale
+
+**Bytecode over Addresses**: Bytecode is deterministic and can be verified on-chain, while addresses are opaque and mutable.
+
+**Reverting on re-indexing**: There is small, yet non-zero probability of hash collision attack. Disallowing updates to indexed location of bytecode coupes with this.
+
+**Simple Interface**: The interface is minimal and focused to maximize composability and ease of implementation.
+
```

Comment:
You should explain where you got your salt from. See https://en.wikipedia.org/wiki/Nothing-up-my-sleeve_number.

Reply:
added explanation